boomerang/src/nicp_to_icp.rs

```diff
@@ -11,6 +11,7 @@ use icrc_ledger_types::icrc1::transfer::TransferArg;
 use icrc_ledger_types::icrc2::approve::ApproveArgs;
 
 pub async fn notify_nicp_deposit(target: Principal) -> Result<WithdrawalSuccess, BoomerangError> {
+    let s = borrow_state();
```

Comment:
This is a problem. I would like you to try to understand why. Deploy everything locally and make two calls to two endpoints simultaneously and see what happens.

Reply:
Discussed on telegram